pkg/statscombiner/statscombiner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statscombiner
+
+import (
+	"sync"
+)
+
+type statsWrapper[T any] struct {
+	stats []T
+	ttl   int
+}
+
+type StatsCombiner[T any] struct {
+	lock  sync.Mutex
+	ttl   int
+	stats map[string]*statsWrapper[T]
+}
+
+// NewStatsCombiner creates a new stats combiner that stores structs of type T using a key. Each key is given a
+// time-to-live (ttl) and only stored for ttl calls of GetStats(). Whenever a key is refreshed using AddStats(),
+// the ttl will be reset to the initial value.
+func NewStatsCombiner[T any](ttl int) *StatsCombiner[T] {
+	return &StatsCombiner[T]{
+		ttl:   ttl,
+		stats: make(map[string]*statsWrapper[T]),
+	}
+}
+
+// AddStats adds the given stats to the given key and set its ttl to the ttl of the StatsCombiner
+func (sc *StatsCombiner[T]) AddStats(key string, stats []T) {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	if curStats, ok := sc.stats[key]; ok {
+		curStats.stats = stats
+		curStats.ttl = sc.ttl
+		return
+	}
+	sc.stats[key] = &statsWrapper[T]{
+		stats: stats,
+		ttl:   sc.ttl,
+	}
```

Comment:
You're right - those are the same. Initially I thought it would be a good idea to count the number of AddStats() calls as well, which required to differentiate the cases. I will remove that block. 👍